File: app/src/main/java/io/intrepid/skotlinton/screens/example1/Example1Contract.java
package io.intrepid.skotlinton.screens.example1;

import io.intrepid.skotlinton.base.BaseContract;

class Example1Contract {
    public interface View extends BaseContract.View {

        void gotoExample2();
    }

    public interface Presenter extends BaseContract.Presenter<View> {

        void onButtonClick();
    }
}

File: app/src/main/java/io/intrepid/skotlinton/screens/example1/Example1Contract.kt
package io.intrepid.skotlinton.screens.example1

import io.intrepid.skotlinton.base.BaseContract

internal class Example1Contract {
    interface View : BaseContract.View {

        fun gotoExample2()
    }

    interface Presenter : BaseContract.Presenter<View> {

        fun onButtonClick()
    }
}
